nixos/systemd-boot: add efiBootOptionDescription option

kurtkilpela opened the pull request:
## Description of changes
By default, `bootctl` will use "Linux Boot Manager" as the boot option description. This is not ideal when dual-booting. In systemd 252, the `--efi-boot-option-description` flag was added to permit custom description text.

When `boot.loader.systemd-boot.efiBootOptionDescription` option is not set, the `--efi-boot-option-description` is not used and `bootctl` is left to use its default value.


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

JulienMalka commented on nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix:
```diff
@@ -38,7 +38,9 @@ let
-    inherit (cfg) consoleMode graceful editor;
+    inherit (cfg) consoleMode graceful editor efiBootOptionDescription;
+
+    setEfiBootOptionDescription = cfg.efiBootOptionDescription != null;
```
Why not do this test on the python side?

kurtkilpela replied:
I wasn't sure of how to do that reliably. If `efiBootOptionDescription` is `null`, the substitution logic will not change the value. I could recreate the "@efiBootOptionDescription@" through string concatenation to do the comparison. It seemed a little odd so I did it this way.

Frontear commented on maintainers/maintainer-list.nix:
Separate this commit from rest of commits

kurtkilpela replied:
Thank you. Fixed.

Frontear commented on nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix:
```diff
@@ -317,6 +319,21 @@ in {
+        Invoke `bootctl` with the `--efi-boot-option-description` option,
+        which specifies the boot entry name.
+
+        This option allows a user to distringuish between separate bootloaders
```
distinguish

kurtkilpela replied:
Fixed.

Commit:
maintainers: add kurtkilpela

Commit:
nixos/systemd-boot: add efiBootOptionDescription option

By default, `bootctl` will use "Linux Boot Manager" as the boot option description. This is not ideal when dual-booting. In systemd 252, the `--efi-boot-option-description` flag was added to permit custom description text.

When `boot.loader.systemd-boot.efiBootOptionDescription` option is not set, the `--efi-boot-option-description` is not used and `bootctl` is left to use its default value.

kurtkilpela commented:
The commit message has been updated to align with the nixos guidelines.
